legends/load_legends.py

```diff
@@ -0,0 +1,309 @@
+                    breaks_min = min(breaks_min, np.min(blocks_data))
+                    breaks_max = max(breaks_max, np.max(blocks_data))
+
+            # computing breaks, considering all rasters
+            breaks = np.percentile(total_sample_data, percentiles_list)
+
+        else:
+            print(
+                'Computing legend breaks - dataset {}, layer "{}"\n'.format(
+                    ds_id, variable
+                )
+            )
+            vector = gdf
+            vector["variable"] = vector["variables"].apply(
+                lambda x: list(x.values())[0]
+            )
+            data = np.array(vector["variable"])
+
+            breaks = np.percentile(data, percentiles_list)
+            breaks_min = np.min(data)
+            breaks_max = np.max(data)
+
+        # round and export unique breaks
+        breaks_output = np.append(breaks_min, breaks)
+        breaks_output = np.append(breaks_output, breaks_max)
+        breaks_output = np.unique(np.around(breaks_output, decimals=2))
+
+        if np.sum(breaks_output) == 0:
+            raise ValueError("All the breaks of the layer are 0", ds_id, variable, fid)
+        if len(breaks_output) < 2:
+            raise ValueError("The breaks are less than 2")
+
+        with open(layer_path, "w") as f:
+            np.savetxt(f, breaks_output)
+
+    else:
+        print('No data found for dataset\'s {} layer "{}"'.format(ds_id, variable))
+
+# ------------------------------
+# loading legends
+# ------------------------------
+
+for i, row in layers.iterrows():
+
+    if i not in EXCLUDED:
+        ds_id = row["ds_id"]
+        variable = row["variable"]
+        valid_fname = format_filename(variable)
+        data_class = row["class_of_data"]
+        clrs = colors_dict[data_class]
+        layer_path = BREAKSDIR / "{}_{}.txt".format(ds_id, valid_fname)
+
+        if os.path.exists(layer_path):
+            print("Uploading legend - dataset {} - {}".format(ds_id, variable))
+            legend = {"name": variable, "type": "custom", "symbology": []}
+            breaks = np.loadtxt(layer_path)
+
+            # prepare the legend without considering the last break
+            for color_index in range(len(breaks) - 1):
+                red = int(clrs[color_index][0] * 255)
+                green = int(clrs[color_index][1] * 255)
+                blue = int(clrs[color_index][2] * 255)
+                legend["symbology"].append(
+                    {
+                        "red": red,
+                        "green": green,
+                        "blue": blue,
+                        "opacity": 1.0,
+                        "value": breaks[color_index],
+                        "label": "≥ "
+                        + str(round(breaks[color_index], 2))
+                        + " "
+                        + str(row["unit"]),
+                    }
+                )
+
+            # send the legend
+            r = requests.post(
```

Comment:
Since we connect anyway to the database service, I think it would be better to use the `enermaps_set_legend` directly as a SQL query. It should be faster and "cleaner". Maybe to be added to the README as something that can be considered by future users that implement their own EnerMaps.